main/typings.d.ts

```diff
@@ -11,12 +11,52 @@ export interface Activity {
 }
 
 export type FILTransaction = {
-  hash: string
-  timestamp: number
-  status: TransactionStatus
-  outgoing: boolean
-  amount: string
-  address: string
+  hash: string;
+  height: number;
+  timestamp: number;
+  status: TransactionStatus;
+  outgoing: boolean;
+  amount: string;
+  address: string;
+}
+
+// helper
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>
+
+// FILTransaction with certain properties changed to optional
+export type FILTransactionProcessing = PartialBy<FILTransaction, 'hash' | 'height'>
```

Comment:
IIUC, we use this type when the transaction has `status` set to `processing`. You may want to encode that fact in the type too.

```suggestion
export type FILTransactionProcessing = PartialBy<FILTransaction, 'hash' | 'height'> & { status: 'processing' };
```

Reply:
<img width="556" alt="Screenshot 2023-01-30 at 15 10 32" src="https://user-images.githubusercontent.com/10247/215500252-0e3be6c1-2b06-4446-a143-271db71b13ae.png">

If a transaction starts as processing and then is market as failed, how can I convert the type? Or shall I recreate the transaction object?